src/main.rs

```diff
@@ -107,8 +118,79 @@ fn main() -> Result<()> {
     }
 }
 
+fn setup_logging() -> Result<WorkerGuard> {
+    use tracing::metadata::LevelFilter;
+    use tracing::Level;
+    use tracing_subscriber::prelude::*;
+    use tracing_subscriber::{
+        field::RecordFields,
+        fmt::{
+            self,
+            format::{Pretty, Writer},
+            FormatFields,
+        },
+        EnvFilter,
+    };
+
+    /// Workaround for <https://github.com/tokio-rs/tracing/issues/1817>.
+    struct NewType(Pretty);
+
+    impl<'writer> FormatFields<'writer> for NewType {
+        fn format_fields<R: RecordFields>(
+            &self,
+            writer: Writer<'writer>,
+            fields: R,
+        ) -> core::fmt::Result {
+            self.0.format_fields(writer, fields)
+        }
+    }
+
+    let project_dirs =
+        ProjectDirs::from("", "", "drg-mod-integration").context("constructing project dirs")?;
+    std::fs::create_dir_all(project_dirs.data_dir())?;
+
+    let f = File::create(project_dirs.data_dir().join("drg-mod-integration.log"))?;
+    let writer = BufWriter::new(f);
+    let (log_file_appender, guard) = tracing_appender::non_blocking(writer);
+    let debug_file_log = fmt::layer()
+        .with_writer(log_file_appender)
+        .fmt_fields(NewType(Pretty::default()))
+        .with_ansi(false)
+        .with_filter(filter::filter_fn(|metadata| {
+            *metadata.level() <= Level::DEBUG && metadata.target() == "drg_mod_integration"
+        }));
```

Comment:
This seems to filter only messages matching exactly `drg_mod_integration`. I think this is what you want:
```rust
        .with_filter(
            tracing_subscriber::filter::Targets::new()
                .with_target("drg_mod_integration", Level::DEBUG),
        );
```